### PR TITLE
Add missing DataBlock-releated functions to the C-API

### PR DESCRIPTION
**Pull request by dvdplm**
Adds two missing functions to the C-API:

- `rocksdb_block_based_options_set_data_block_index_type`
- `rocksdb_block_based_options_set_data_block_hash_ratio`

This enables users in other languages to enjoy the new(-ish) feature.

The changes here are partially overlapping with [another PR](https://github.com/facebook/rocksdb/pull/5630) but are more focused on the DataBlock indexing options.

### CODE REVIEW

**Comment by siying** [comment]
Can you add a unit test, just calling new functions?

**Comment by dvdplm** [comment]
> Can you add a unit test, just calling new functions?

[Done](https://github.com/facebook/rocksdb/pull/6101/commits/05f75dfd1637f8f60e1eec5d9e1b4923839a5f4e), I think. :)

File: db/c.cc [hunk 1]
```diff
@@ -2032,6 +2032,16 @@ void rocksdb_block_based_options_set_index_type(
   options->rep.index_type = static_cast<BlockBasedTableOptions::IndexType>(v);
 }
 
+void rocksdb_block_based_options_set_data_block_index_type(
+    rocksdb_block_based_table_options_t* options, int v) {
+  options->rep.data_block_index_type = static_cast<BlockBasedTableOptions::DataBlockIndexType>(v);
```
Comment (siying) [inline db/c.cc]: This is over 80-char line limit.

**Comment by dvdplm** [comment]
@siying the CI failures seem spurious but I'm not very familiar with the testing setup here.

**Comment by dvdplm** [comment]
Can someone help me figure out which released version contains this change?

**Comment by pdillinger** [comment]
> Can someone help me figure out which released version contains this change?

6.7.x

**Pull request closed without merging**